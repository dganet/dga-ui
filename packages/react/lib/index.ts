import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}