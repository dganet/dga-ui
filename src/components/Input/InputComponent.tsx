import { useTheme } from '@hooks/useTheme'
import React, { BaseSyntheticEvent, useState } from 'react'

import {
    Container,
    InputLabel,
    StyledInput,
    InputContainer,
    StyledButton
} from './InputComponent.styles'
import { InputProps } from './InputComponent.types'
import { AiOutlinePlus, AiOutlineMinus } from 'react-icons/ai'
const InputComponent = ({
    label,
    inputSize = 'large',
    inputType = 'regular',
    inputIconSide = 'left',
    icon: Icon = null,
    error = false,
    ...props
}: InputProps) => {
    const theme = useTheme()
    const [valNum, setValNum] = useState<number>(0)

    const plusClick = () => setValNum(valNum + 1)
    const minusClick = () => valNum > 0 && setValNum(valNum - 1)
    const handleNumberChange = (e: BaseSyntheticEvent) => {
        try {
            const value = parseInt(e.currentTarget.value)
            setValNum(value)
        } catch {}
    }

    return (
        <Container inputSize={inputSize} inputType={inputType}>
            <InputLabel theme={theme}>{label}</InputLabel>
            <InputContainer
                theme={theme}
                inputSize={inputSize}
                inputType={inputType}
                error={error}
            >
                {inputType === 'number' ? (
                    <>
                        <StyledButton onClick={plusClick}>
                            <AiOutlinePlus />
                        </StyledButton>
                        <StyledInput
                            inputSize={inputSize}
                            inputType={inputType}
                            theme={theme}
                            type="text"
                            value={valNum}
                            onChange={handleNumberChange}
                            {...props}
                        />
                        <StyledButton onClick={minusClick}>
                            <AiOutlineMinus />
                        </StyledButton>
                    </>
                ) : (
                    <>
                        {inputIconSide === 'left' && Icon && (
                            <Icon style={{ marginRight: '8px' }} />
                        )}
                        <StyledInput theme={theme} type="text" {...props} />
                        {inputIconSide === 'right' && Icon && <Icon />}
                    </>
                )}
            </InputContainer>
        </Container>
    )
}

export default InputComponent
