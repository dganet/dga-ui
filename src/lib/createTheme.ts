import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
