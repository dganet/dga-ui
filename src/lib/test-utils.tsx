import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
