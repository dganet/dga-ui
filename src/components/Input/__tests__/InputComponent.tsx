import { render, screen } from '../../../lib/test-utils'

import InputComponent from '../InputComponent'

describe('<InputComponent />', () => {
    it('should render the component', () => {
        const { container } = render(<InputComponent label="test" />)

        expect(container.firstChild).toMatchSnapshot()
    })
})
