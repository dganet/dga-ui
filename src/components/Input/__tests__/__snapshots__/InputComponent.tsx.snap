// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent /> should render the component 1`] = `
<div
  class="sc-bdfBQB aujAO"
>
  <label
    class="sc-gsTEea itrQNo"
  >
    test
  </label>
  <div
    class="sc-dlfnuX bkBHeL"
  >
    <input
      class="sc-hKgJUU cSydUb"
      type="text"
    />
  </div>
</div>
`;
